### DISCUSSION TITLE
How to generate Scaladoc for multiple Scala versions?

### DISCUSSION THREAD

**Discussion by pathikrit** [sbt/sbt#7151] [Q&A]
What did I try?
I tried this:
```
  .enablePlugins(SiteScaladocPlugin, PreprocessPlugin)
  .settings(
    SiteScaladoc / siteSubdirName := "api" + "/" + scalaVersion.value,
  )
```

Then I did `sbt +makeSite` but only the latest version would remain since previous ones would get cleaned up...

I never figured this out so I rolled my own.

See: https://pathikrit.github.io/better-files/

Code:
https://github.com/pathikrit/better-files/blob/3782a29b8d4b6be049f24bfa8458a6ee243d361f/build.sbt#L40

**Comment by MazieRamyqfW** [comment]
It sounds like you were trying to customize the output directory of Scaladoc when generating a website using the SiteScaladocPlugin in SBT. However, you were experiencing issues with previous versions getting cleaned up and were not able to figure out the solution.

It's great that you found a solution by rolling your own implementation and using the better-files library. The better-files library provides a convenient and intuitive API for working with files and directories in Scala.

If you still want to use the SiteScaladocPlugin in SBT and customize the output directory, you may want to try the following:

.enablePlugins(SiteScaladocPlugin)
.settings(
  SiteScaladoc / siteSubdirName := s"api/${scalaBinaryVersion.value}/${scalaVersion.value}",
  makeSite / target := baseDirectory.value / "site"
)

This sets the siteSubdirName to api/[binary version]/[scala version], and then sets the target directory of the makeSite task to be the site directory in the project root.

**Reply by pathikrit** [reply]
Looks like this was written by ChatGPT ...

> It's great that you found a solution by rolling your own implementation and using the better-files library. The better-files library provides a convenient and intuitive API for working with files and directories in Scala.

Umm ... this is the `build.sbt` _of_ the better-files project and does not use better-files itself!

The following suggestion sounds good in theory:
```scala
SiteScaladoc / siteSubdirName := s"api/${scalaBinaryVersion.value}/${scalaVersion.value}",
```

But, in practice when `+makeSite` is called, the whole top-level `api` directory is nuked for each `scalaVersion` thus, only the ScalaDoc remains for the last `scalaBinaryVersion` ...

**Reply by SethTisue** [reply]
I agree that this is AI generated spam. Note that it's from a brand new account. I reported the account to GitHub for spamming and I encourage others to do the same.

I have seen similar in other repos I am watching.
